src/Map/Map.cpp

```diff
@@ -450,8 +450,34 @@ void Map::resetMapInstance() {
         delete map_instance;
         map_instance = nullptr;
     }
+    std::cout << "resetMapInstance( ) works." << std::endl;
 }
 
+Map *Map::getInstance() {
+    // generate map
+    bool validMap = false;
+    // will run until a valid map is provided
+    while (!validMap) {
+        try {
+            // get file location
+            std::string fileLocation;
+            std::cout << "Enter map file location: ";
+            std::cin >> fileLocation;
+
+            // load map
+            MapLoader loader;
+            map_instance = loader.generateMap(fileLocation);
+            validMap = true;
+        }
+        catch (const char *msg) {
+            std::cout << "Error: Invalid map file. ";
+            std::cout << msg << std::endl;
+        }
+    }
```

Comment:
This good, the only issue is that this will make a new map every time you run it. Maybe put this behind an `if` statement like:

```
if (map_instance == 0) {
    // your code moves in here
}
```